app/models/sample.rb

```diff
@@ -230,7 +230,7 @@ class Metadata
                 with: /\A[[:ascii:]]+\z/,
                 message: 'only allows ASCII'
               },
-              if: :supplier_name_changed?
+              if: :supplier_name_changed? && :supplier_name?
```

Comment:
Does this automatiicaly trigger the `touch` which resends the sample to the MLWH?

Reply:
Hopefully it should be sent to the warehouse when we update to nil. Didn't test it locally though so I will def check this in UAT.

Reply:
Tested this in UAT - updates MLWH fine when filling in and blanking supplier_name